modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ImageStorage.java

```diff
@@ -380,7 +380,20 @@ public ImageFrame[] loadAll(String input, ImageLoadListener listener,
                     }
 
                     if (theStream == null) {
-                        theStream = ImageTools.createInputStream(input);
+                        try {
+                            theStream = ImageTools.createInputStream(input);
+                        } catch (IOException ignored) {
+                        }
+                    }
+
+                    if (theStream == null) {
+                        try {
+                            // last fallback, try to see if the file exists with @1x suffix
+                            String scaled1xName = ImageTools.getScaledImageName(input, 1);
```

Comment:
I think this function does not properly handle cases where the input may already have a scale specified.

If I want to load `testimg@1x.png` then this function will return `testimg@1x@1x.png`

I'm not sure what is normal here, but I'd think that if a scale was already specified that we shouldn't do any fallback.

Reply:
This technically translates to the for-loop higher up that tries to fetch other scaled versions of an image - if you want to explicitly load `foo@1x.png` it will look for `foo@1x@2x.png` etc.

Should we just assume that with `@Nx` provided in the name we will only load that one specific scale?

Reply:
I would say so, yes. If the user specifies any `@Nx`, then we only look for this particular version.

Reply:
Probably, but this is out of scope for this PR / JBS bug. Please file a follow-up bug.

Reply:
Filed [JDK-8342530](https://bugs.openjdk.org/browse/JDK-8342530)